src/jdk.jdwp.agent/share/native/libjdwp/threadControl.c

```diff
@@ -412,14 +412,19 @@ insertThread(JNIEnv *env, ThreadList *list, jthread thread)
             if (error != JVMTI_ERROR_NONE) {
                 EXIT_ERROR(error, "getting vthread state");
             }
+            if ((vthread_state & JVMTI_THREAD_STATE_ALIVE) == 0) {
+              // Thread not alive so put on otherThreads list instead of runningVThreads.
+              // It might not have started yet or might have terminated. Either way,
+              // otherThreads is the place for it.
```

Comment:
If a terminated thread is added to otherThreads then will be it eventually removed?

Reply:
Yes, I had the same question. otherThreads is swept every time any thread is resumed. This code really isn't much different than what is done for platform threads, except that the detection that it should go on otherThreads is done earlier on commonSuspend(). The code differences arise from the differences in how platform threads and virtual threads are tracked. I hope to unify them someday, but for now don't want virtual thread support to risk destabilizing platform thread support.